class_configs/brd_class_config.lua

```diff
@@ -40,23 +50,168 @@ local Tooltips     = {
+        addSong('WarMarchSong')
+        addSong('MainAriaSong')
+        ConditionallyAddSong('UseProgressive', 'DichoSong')
+    end
+
+    local function AddSelfDPSSongs()
+        ConditionallyAddSong("UseAlliance", "AllianceSong")
+        ConditionallyAddSong("UseInsult", 'InsultSong')
+        ConditionallyAddSong("UseFireDots", "FireDotSong")
+        ConditionallyAddSong("UseIceDots", "IceDotSong")
+        ConditionallyAddSong("UseDiseaseDots", "DiseaseDotSong")
+        ConditionallyAddSong("UsePoisonDots", "PoisonDotSong")
+    end
+
+    local function AddMeleeDPSSongs()
+        ConditionallyAddSong("UseSuffering", "SufferingSong")
+    end
+
+    local function AddTankSongs()
+        ConditionallyAddSong("UseSpiteful", "SpitefulSong")
+        ConditionallyAddSong("UseSpry", "SprySonataSong")
+        ConditionallyAddSong("UseResist", "ResistSong")
+    end
+
+    local function AddHealerSongs()
+        ConditionallyAddSong("UseReckless", "RecklessSong")
+        ConditionallyAddSong("UsePulse", "PulseRegenSong")
+    end
+
+    local function AddCasterDPSSongs()
+        ConditionallyAddSong("UseArcane", "ArcaneSong")
+        ConditionallyAddSong("UseFireDDMod", "CasterAriaSong")
+        ConditionallyAddSong("UseFate", "FateSong")
+        ConditionallyAddSong("UsePotency", "PotencySong")
+    end
+
+    local function AddRegenSongs()
+        ConditionallyAddSong("UseAmp", "AmpSong")
+        ConditionallyAddSong("UseCrescendo", "CrescendoSong")
+        ConditionallyAddSong("UseChorus", "ChorusRegenSong")
+        ConditionallyAddSong("UseCantata", "CantataRegenSong")
+    end
+    -----------------------------------------------------------------------------------------
+    local mode = RGMercUtils.GetSetting("Mode")
+    AddCriticalSongs()
+    if mode == 1 then -- General
+        AddMainGroupDPSSongs()
+        AddSelfDPSSongs()
+        AddRegenSongs()
+        AddMeleeDPSSongs()
+        AddCasterDPSSongs()
+        AddHealerSongs()
+        AddTankSongs()
+    elseif mode == 2 then -- Tank
+        AddTankSongs()
+        AddMainGroupDPSSongs()
+        AddHealerSongs()
+        AddMeleeDPSSongs()
+        AddRegenSongs()
+        AddSelfDPSSongs()
+        AddCasterDPSSongs()
+    elseif mode == 3 then -- Caster
+        AddMainGroupDPSSongs()
+        AddCasterDPSSongs()
+        AddRegenSongs()
+        AddSelfDPSSongs()
+        AddMeleeDPSSongs()
+        AddHealerSongs()
+        AddTankSongs()
+    elseif mode == 4 then -- Healer
+        AddHealerSongs()
+        AddMainGroupDPSSongs()
+        AddRegenSongs()
+        AddSelfDPSSongs()
+        AddCasterDPSSongs()
+        AddTankSongs()
+        AddMeleeDPSSongs()
+    else
+        RGMercsLogger.log_warning("Bard Mode not found!  Adding DPS songs, but you should select a mode.")
+        AddMainGroupDPSSongs()
+        AddSelfDPSSongs()
+        AddRegenSongs()
+        AddMeleeDPSSongs()
+        AddCasterDPSSongs()
+    end
+
+    if songCount > 13 then
```

Comment:
Why 13? Please don't use magic numbers. Based on the intent I think you actually want to check mq.TLO.Me.NumGems() because on TLP you might only have 8 gems, etc.

Reply:
Yup you're correct, that's much saner